random CI failures, unidentified reason

One CI failed repeatedly, here:

```py
        assert not adsimdet.cam.is_busy
    
>       assert plugin.full_file_name.get().strip() not in ("", "/")
E       AssertionError: assert '/' not in ('', '/')
E        +  where '/' = <built-in method strip of str object at 0x7fe5378fbeb0>()
E        +    where <built-in method strip of str object at 0x7fe5378fbeb0> = '/'.strip
E        +      where '/' = <bound method EpicsSignalBase.get of EpicsSignalRO(read_pv='ad:TIFF1:FullFileName_RBV', name='adsimdet_tiff1_full_file_name', parent='adsimdet_tiff1', value='/', timestamp=1660149768.595167, auto_monitor=True, string=True)>()
E        +        where <bound method EpicsSignalBase.get of EpicsSignalRO(read_pv='ad:TIFF1:FullFileName_RBV', name='adsimdet_tiff1_full_file_name', parent='adsimdet_tiff1', value='/', timestamp=1660149768.595167, auto_monitor=True, string=True)> = EpicsSignalRO(read_pv='ad:TIFF1:FullFileName_RBV', name='adsimdet_tiff1_full_file_name', parent='adsimdet_tiff1', value='/', timestamp=1660149768.595167, auto_monitor=True, string=True).get
E        +          where EpicsSignalRO(read_pv='ad:TIFF1:FullFileName_RBV', name='adsimdet_tiff1_full_file_name', parent='adsimdet_tiff1', value='/', timestamp=1660149768.595167, auto_monitor=True, string=True) = AD_EpicsFileNameTIFFPlugin(prefix='ad:TIFF1:', name='adsimdet_tiff1', parent='adsimdet', read_attrs=[], configuration_attrs=[]).full_file_name

/home/runner/work/apstools/apstools/apstools/devices/tests/test_area_detector_support.py:204: AssertionError
=========================== short test summary info ============================
FAILED apstools/devices/tests/test_area_detector_support.py::test_bp_count_custom_name[tiff1-AD_EpicsFileNameTIFFPlugin]
============ 1 failed, 385 passed, 54 skipped in 186.23s (0:03:06) =============
```

This happens occasionally, that one of the file writing plugins (HDF5, JPEG, or TIFF) does not provide a full file name.  Why?  What makes this fail?  What makes this fail _randomly_?

_Originally posted by @prjemian in https://github.com/BCDA-APS/apstools/issues/681#issuecomment-1211028883_

Yes.  These lines show that:

```py
>       assert plugin.full_file_name.get().strip() not in ("", "/")
E       AssertionError: assert '/' not in ('', '/')
```

which means `plugin.full_file_name.get().strip()` is `/`

[Again](https://github.com/BCDA-APS/apstools/runs/7796544002?check_suite_focus=true), JPEG plugin this time:

```py
        assert not adsimdet.cam.is_busy
    
>       assert plugin.full_file_name.get().strip() not in ("", "/")
E       AssertionError: assert '/' not in ('', '/')
E        +  where '/' = <built-in method strip of str object at 0x7ffa5d9ffeb0>()
E        +    where <built-in method strip of str object at 0x7ffa5d9ffeb0> = '/'.strip
E        +      where '/' = <bound method EpicsSignalBase.get of EpicsSignalRO(read_pv='ad:JPEG1:FullFileName_RBV', name='adsimdet_jpeg1_full_file_name', parent='adsimdet_jpeg1', value='/', timestamp=1660255405.667151, auto_monitor=True, string=True)>()
E        +        where <bound method EpicsSignalBase.get of EpicsSignalRO(read_pv='ad:JPEG1:FullFileName_RBV', name='adsimdet_jpeg1_full_file_name', parent='adsimdet_jpeg1', value='/', timestamp=1660255405.667151, auto_monitor=True, string=True)> = EpicsSignalRO(read_pv='ad:JPEG1:FullFileName_RBV', name='adsimdet_jpeg1_full_file_name', parent='adsimdet_jpeg1', value='/', timestamp=1660255405.667151, auto_monitor=True, string=True).get
E        +          where EpicsSignalRO(read_pv='ad:JPEG1:FullFileName_RBV', name='adsimdet_jpeg1_full_file_name', parent='adsimdet_jpeg1', value='/', timestamp=1660255405.667151, auto_monitor=True, string=True) = AD_EpicsFileNameJPEGPlugin(prefix='ad:JPEG1:', name='adsimdet_jpeg1', parent='adsimdet', read_attrs=[], configuration_attrs=[]).full_file_name

/home/runner/work/apstools/apstools/apstools/devices/tests/test_area_detector_support.py:196: AssertionError
=========================== short test summary info ============================
FAILED apstools/devices/tests/test_area_detector_support.py::test_bp_count_custom_name[jpeg1-AD_EpicsFileNameJPEGPlugin]
============ 1 failed, 385 passed, 54 skipped in 202.83s (0:03:22) =============
```

[Another failure](https://github.com/BCDA-APS/apstools/runs/7796803404?check_suite_focus=true).  It passed the new diagnostic tests then failed the same test it has failed before:

```py
        assert plugin.num_captured.get() == NUM_FRAMES
    
        # diagnostic per issue #696
        ffname = plugin.full_file_name.get().strip()
        if ffname in ("", "/"):
            assert plugin.dropped_arrays.get() == 0
            assert plugin.array_counter.get() == NUM_FRAMES
>       assert ffname not in ("", "/"), str(ffname)
E       AssertionError: /
E       assert '/' not in ('', '/')

/home/runner/work/apstools/apstools/apstools/devices/tests/test_area_detector_support.py:206: AssertionError
FAILED apstools/devices/tests/test_area_detector_support.py::test_bp_count_custom_name[hdf1-AD_EpicsFileNameHDF5Plugin]
```

But also failed a different test:

```py
        value = getattr(gp_info, attr).get()
        if comparison == "start":
            assert value.startswith(expected)
        elif comparison == "end":
>           assert value.endswith(expected)
E           AssertionError: assert False
E            +  where False = <built-in method endswith of str object at 0x7f4a96f08030>('iocxxx')
E            +    where <built-in method endswith of str object at 0x7f4a96f08030> = ''.endswith

/home/runner/work/apstools/apstools/apstools/synApps/tests/test_iocstats.py:47: AssertionError
FAILED apstools/synApps/tests/test_iocstats.py::test_running[application_directory-iocxxx-end]
```

Another failure seen before:

```py
    def test_running(attr, expected, comparison):
        gp_info = IocStatsDevice(IOC, name="gp_info")
        gp_info.wait_for_connection()
        assert gp_info.connected
        short_delay_for_EPICS_IOC_database_processing()
    
        assert hasattr(gp_info, attr)
        value = getattr(gp_info, attr).get(use_monitor=False)
        if comparison == "start":
            assert value.startswith(expected)
        elif comparison == "end":
>           assert value.endswith(expected)
E           AssertionError: assert False
E            +  where False = <built-in method endswith of str object at 0x7fb99e87c030>('iocxxx')
E            +    where <built-in method endswith of str object at 0x7fb99e87c030> = ''.endswith

/home/runner/work/apstools/apstools/apstools/synApps/tests/test_iocstats.py:47: AssertionError
=========================== short test summary info ============================
FAILED apstools/synApps/tests/test_iocstats.py::test_running[application_directory-iocxxx-end]
============ 1 failed, 385 passed, 54 skipped in 236.80s (0:03:56) =============
```

status returned before expected delay interval:

```
=================================== FAILURES ===================================
_________________________ test_move_and_stopped_early __________________________

rbv = EpicsSignal(read_pv='gp:gp:float1', name='rbv', value=3.37, timestamp=1663681204.770351, tolerance=0.0001, auto_monitor=True, string=False, write_pv='gp:gp:float1', limits=False, put_complete=False)
pos = PVPositionerSoftDoneWithStop(prefix='gp:gp:', name='pos', settle_time=0.0, timeout=None, read_attrs=['readback', 'setpoint'], configuration_attrs=['done', 'tolerance', 'target'], limits=None, egu='')

    def test_move_and_stopped_early(rbv, pos):
        short_delay_for_EPICS_IOC_database_processing()
    
        # first, move to some random, non-zero, initial position
        target = round(2 + 5 * random.random(), 2)
        delayed_complete(pos, rbv, delay=0.5)
        status = pos.move(target)  # readback set by delayed_complete()
        short_delay_for_EPICS_IOC_database_processing()
        assert status.done
        assert status.success
>       assert status.elapsed >= 0.5
E       assert 0.49660348892211914 >= 0.5
E        +  where 0.49660348892211914 = MoveStatus(done=True, pos=pos, elapsed=0.5, success=True, settle_time=0.0).elapsed

/home/runner/work/apstools/apstools/apstools/devices/tests/test_positioner_soft_done.py:234: AssertionError
=============================== warnings summary ===============================
../../../micromamba-root/envs/anaconda-test-env-py-3.9/lib/python3.10/site-packages/setuptools_scm/git.py:135
  /home/runner/micromamba-root/envs/anaconda-test-env-py-3.9/lib/python3.10/site-packages/setuptools_scm/git.py:135: UserWarning: "/home/runner/work/apstools/apstools" is shallow and may cause errors
    warnings.warn(f'"{wd.path}" is shallow and may cause errors')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED apstools/devices/tests/test_positioner_soft_done.py::test_move_and_stopped_early
======= 1 failed, 395 passed, 54 skipped, 1 warning in 210.07s (0:03:30) =======
Error: Process completed with exit code 1.
```

--- a/apstools/devices/tests/test_positioner_soft_done.py
+++ b/apstools/devices/tests/test_positioner_soft_done.py
@@ -231,7 +231,7 @@ def test_move_and_stopped_early(rbv, pos):
     short_delay_for_EPICS_IOC_database_processing()
     assert status.done
     assert status.success
-    assert status.elapsed >= 0.5
+    # assert status.elapsed >= 0.5  # can't assure for initial move
 
     # move that is stopped before reaching the target
     longer_delay = 2

